### PR TITLE
python312Package.sure: drop nose dependency

### PR DESCRIPTION
**Pull request by Sigmanificient**
## Description of changes

Part of:
- #326513

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by emilazy — CHANGES_REQUESTED** [review]
Please migrate to `pyproject = true;` at the same time.

Maybe we should consider bumping this to 3.0a2, which seems to have removed the nose dependency entirely. I don’t know if it has breaking changes or not, though.

**Comment by Sigmanificient** [comment]
Well there isn't any tags for `3.0.1a` and it hasn't yet been updated on pypi too (https://pypi.org/project/sure/#files), so I believe the intent of the developer is to keep working on it before doing the release

File: pkgs/development/python-modules/sure/default.nix [hunk 1]
```diff
@@ -23,17 +23,20 @@ buildPythonPackage rec {
...
     mock
-    six
   ];
-
-  doCheck = pythonOlder "3.12"; # nose requires imp module
-
-  nativeCheckInputs = [ nose ];
+  preCheck = ''
+    rm tests/test_old_api.py # require nose
+  '';
```
Comment (emilazy) [inline pkgs/development/python-modules/sure/default.nix:39]: Can you use `disabledTests`/`disabledTestPaths` here? (Nit: Please add a newline before the definition, as well.)

**Review by emilazy — COMMENTED** [review]
3.0a2 is on PyPI if you look at [the right page](https://pypi.org/project/sure/#history), but it’s fine to wait for a final release.

**Review by emilazy — APPROVED** [review]
I built `python{311,312}Packages.sure` on `x86_64-darwin`, `x86_64-linux`, and `aarch64-linux`. Since thousands of packages depend on this one, I haven’t tested the full tree, but as this is a tests‐only change it shouldn’t propagate further. Retargetting to `staging` for the merge.

**Review by natsukium — APPROVED** [review]
lgtm

**Pull request merged**